waspc/examples/todoApp/ext/Todo.js

```diff
@@ -98,12 +99,25 @@ const Tasks = (props) => {
 }
 
 const Task = (props) => {
+  const action = useAction(updateTaskIsDone, {
+    optimisticUpdates: [{
+      getQuery: () => [getTasks],
+      updateQuery: (updatedTask, oldTasks) => {
+        if (oldTasks === undefined) {
+          // cache is empty
+          return [updatedTask];
+        } else {
+          return oldTasks.map(task => task.id == updatedTask.id ? { ...task, ...updatedTask } : task)
+        }
+      }
+    }]
+  });
   const handleTaskIsDoneChange = async (event) => {
     const id = parseInt(event.target.id)
     const isDone = event.target.checked
 
     try {
-      await updateTaskIsDone({ id, isDone })
+      await action.mutateAsync({ id, isDone })
```

Comment:
One last issue with the counter I forgot to mention.

Since we postpone updating the query as long as it has pending actions, the results of other (unrelated) actions that would affect the same query won't be shown in the UI until the client receives a response to the slowest action. Here's a nice example of this behavior:

https://user-images.githubusercontent.com/22752307/179479457-576e9dda-9222-4a69-b432-f299dd43a89d.mp4

Reply:
Arg, good point.